### PR TITLE
up minimum php version to 7.2 (WP-567)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

**Comment by vsolovei-smartling** [comment]
Current state of tests:
....WWWW.....WR.W............WW............................RRRR  63 / 269 ( 23%)
....W..................................RRR..RRRRRR............. 126 / 269 ( 46%)
..W............................................................ 189 / 269 ( 70%)
..........................................................W.W.. 252 / 269 ( 93%)
.................                                               269 / 269 (100%)

Time: 7.2 minutes, Memory: 64.25 MB

There were 12 warnings:

1) Smartling\Tests\DetectChangesTest::testCheckSubmissionHash with data set #0 (array(1, '', 1, 'abc', 'post', 1, 'any', 'any', 0, 0, 'any', 'any', 'any', 0, 0, 'In Progress', 0, 0), false, 'abc')
Trying to configure method "getProjectId" which cannot be configured because it does not exist, has not been specified, is final, or is static

2) Smartling\Tests\DetectChangesTest::testCheckSubmissionHash with data set #1 (array(1, '', 1, 'abc', 'post', 1, 'any', 'any', 0, 0, 'any', 'any', 'any', 0, 0, 'In Progress', 0, 1), false, 'abc')
Trying to configure method "getProjectId" which cannot be configured because it does not exist, has not been specified, is final, or is static

3) Smartling\Tests\DetectChangesTest::testCheckSubmissionHash with data set #2 (array(1, '', 1, 'abc', 'post', 1, 'any', 'any', 0, 0, 'any', 'any', 'any', 0, 0, 'In Progress', 0, 1), false, 'def')
Trying to configure method "getProjectId" which cannot be configured because it does not exist, has not been specified, is final, or is static

4) Smartling\Tests\DetectChangesTest::testCheckSubmissionHash with data set #3 (array(1, '', 1, 'abc', 'post', 1, 'any', 'any', 0, 0, 'any', 'any', 'any', 0, 0, 'New', 0, 1), true, 'def')
Trying to configure method "getProjectId" which cannot be configured because it does not exist, has not been specified, is final, or is static

5) Smartling\Tests\FileUriHelperTest::testPreparePermalinkInvalidParams with data set #0 ('http://nothing.com/blog/my-so...title/', null)
The @expectedException, @expectedExceptionCode, @expectedExceptionMessage, and @expectedExceptionMessageRegExp annotations are deprecated. They will be removed in PHPUnit 9. Refactor your test to use expectException(), expectExceptionCode(), expectExceptionMessage(), or expectExceptionMessageMatches() instead.

6) Smartling\Tests\Jobs\LastModifiedCheckJobTest::testRun with data set #0 (array(array(array(array(1, 'A', 1, '', 'post', 7, 'FileA', 'LangA', 2, null, '', null, null, 0, 0, 0, 0, 'In Progress', 0, 0, DateTime Object (...)), array(1, 'A', 1, '', 'post', 7, 'FileA', 'LangB', 2, null, '', null, null, 0, 0, 0, 0, 'In Progress', 0, 0, DateTime Object (...)))), false), array(DateTime Object (...), DateTime Object (...)), 0)
Trying to configure method "storeSubmissions" which cannot be configured because it does not exist, has not been specified, is final, or is static

7) Smartling\Tests\Jobs\LastModifiedCheckJobTest::testFailLastModified with data set #0 (array(array(array(array(1, 'A', 1, '', 'post', 7, 'FileA', 'LangA', 2, null, '', null, null, 0, 0, 0, 0, 'In Progress', 0, 0, DateTime Object (...)), array(1, 'A', 1, '', 'post', 7, 'FileA', 'LangB', 2, null, '', null, null, 0, 0, 0, 0, 'In Progress', 0, 0, DateTime Object (...)))), false), 'Array\n(\n    [0] => Array\n ...  )\n)', 1)
Trying to configure method "storeSubmissions" which cannot be configured because it does not exist, has not been specified, is final, or is static

8) Smartling\Tests\Jobs\LastModifiedCheckJobTest::testFailLastModified with data set #1 (array(array(array(array(1, 'A', 1, '', 'post', 7, 'FileA', 'LangA', 2, null, '', null, null, 0, 0, 0, 0, 'In Progress', 0, 0, DateTime Object (...)), array(1, 'A', 1, '', 'post', 7, 'FileA', 'LangB', 2, null, '', null, null, 0, 0, 0, 0, 'In Progress', 0, 0, DateTime Object (...)))), false), 'Array\n(\n    [0] => Array\n ...  )\n)', 0)
Trying to configure method "storeSubmissions" which cannot be configured because it does not exist, has not been specified, is final, or is static

9) Smartling\Tests\QueryBuilder\ConditionTest::testConditionValidation
The @expectedException, @expectedExceptionCode, @expectedExceptionMessage, and @expectedExceptionMessageRegExp annotations are deprecated. They will be removed in PHPUnit 9. Refactor your test to use expectException(), expectExceptionCode(), expectExceptionMessage(), or expectExceptionMessageMatches() instead.

10) Smartling\Tests\Smartling\Base\SmartlingCoreTest::testExceptionOnTargetPlaceholderCreationFail
Trying to configure method "getFunctionProxyHelper" which cannot be configured because it does not exist, has not been specified, is final, or is static

11) Smartling\Tests\Smartling\Settings\SettingsManagerTest::testGetProfileTargetBlogIdsByMainBlogIdWithDbException
The @expectedException, @expectedExceptionCode, @expectedExceptionMessage, and @expectedExceptionMessageRegExp annotations are deprecated. They will be removed in PHPUnit 9. Refactor your test to use expectException(), expectExceptionCode(), expectExceptionMessage(), or expectExceptionMessageMatches() instead.

12) Smartling\Tests\Smartling\Settings\SettingsManagerTest::testGetProfileTargetBlogIdsByMainBlogIdWithConfigException
The @expectedException, @expectedExceptionCode, @expectedExceptionMessage, and @expectedExceptionMessageRegExp annotations are deprecated. They will be removed in PHPUnit 9. Refactor your test to use expectException(), expectExceptionCode(), expectExceptionMessage(), or expectExceptionMessageMatches() instead.

--

There were 14 risky tests:

1) Smartling\Tests\Jobs\AbstractJobTest::testRunCronJobUnknownSource
This test did not perform any assertions

/plugin-dir/tests/Jobs/AbstractJobTest.php:32

2) Smartling\Tests\PostEntityStdTest::testGetMetadata with data set #0 (array(array('value')), array('value'))
This test did not perform any assertions

/plugin-dir/tests/PostEntityStdTest.php:68

3) Smartling\Tests\PostEntityStdTest::testGetMetadata with data set #1 (array(array('a', 'b', 'c')), array(array('a', 'b', 'c')))
This test did not perform any assertions

/plugin-dir/tests/PostEntityStdTest.php:68

4) Smartling\Tests\PostEntityStdTest::testGetMetadata with data set #2 (array(array('s'), array('a', 'd')), array('s', array('a', 'd')))
This test did not perform any assertions

/plugin-dir/tests/PostEntityStdTest.php:68

5) Smartling\Tests\PostEntityStdTest::testGetMetadataDataProvider
This test did not perform any assertions

/plugin-dir/tests/PostEntityStdTest.php:86

6) Smartling\Tests\Queue\QueueTest::testEnqueue with data set #0 ('upload', array(1), 'INSERT IGNORE INTO `smartling...9a0e')')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:95

7) Smartling\Tests\Queue\QueueTest::testEnqueue with data set #1 ('upload', array(1, 2, 3), 'INSERT IGNORE INTO `smartling...c2b7')')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:95

8) Smartling\Tests\Queue\QueueTest::testEnqueue with data set #2 ('upload', array(stdClass Object ()), 'INSERT IGNORE INTO `smartling...9b1e')')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:95

9) Smartling\Tests\Queue\QueueTest::testDequeue with data set #0 ('upload', 'SELECT `id`, `queue`, `payloa...IT 0,1')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:204

10) Smartling\Tests\Queue\QueueTest::testDequeue with data set #1 ('download', 'SELECT `id`, `queue`, `payloa...IT 0,1')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:204

11) Smartling\Tests\Queue\QueueTest::testPurge with data set #0 ('upload', 'DELETE FROM `smartling_queue`...oad' )')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:227

12) Smartling\Tests\Queue\QueueTest::testPurge with data set #1 ('download', 'DELETE FROM `smartling_queue`...oad' )')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:227

13) Smartling\Tests\Queue\QueueTest::testPurge with data set #2 (null, 'DELETE FROM `smartling_queue`')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:227

14) Smartling\Tests\Queue\QueueTest::testStats with data set #0 ('SELECT `queue`, count(`id`) A...queue`')
This test did not perform any assertions

/plugin-dir/tests/Queue/QueueTest.php:253

WARNINGS!
Tests: 269, Assertions: 436, Warnings: 12, Risky: 14.

File: inc/Smartling/Helpers/FileUriHelper.php [hunk 1]
```diff
@@ -7,84 +7,53 @@
...
-    {
-        if (!($submission instanceof SubmissionEntity)) {
-            throw new InvalidArgumentException('Expected SubmissionEntity');
-        }
-
-        if (StringHelper::isNullOrEmpty($submission->getSourceTitle(false))) {
-            $submission->setSourceTitle('UNTITLED');
-            //throw new InvalidArgumentException('sourceTitle cannot be empty.');
-        }
-    }
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/FileUriHelper.php:39]: I expected only php version related changes. Now we don't throw `throw new InvalidArgumentException('Expected SubmissionEntity');`

File: inc/Smartling/Helpers/WordpressContentTypeHelper.php [hunk 2]
```diff
@@ -38,7 +38,7 @@ private static function getDynamicReverseMap()
         return $map;
     }
 
-    public static $internalTypes = [];
+    public static $internalTypes = ['post' => 'post'];
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/WordpressContentTypeHelper.php:41]: What issue does this change fix? let's not mix refactoring with fixing issues with refactoring and with other tasks

File: inc/Smartling/Queue/Queue.php [hunk 3]
```diff
@@ -179,7 +179,7 @@ private function get($queue)
                 'page'  => 1,
             ]);
 
-        $result = $this->getDbal()->fetch($query, \ARRAY_A);
```
Comment (PavelLoparev) [inline inc/Smartling/Queue/Queue.php:182]: Looks like `\ARRAY_A` is a constant?

File: readme.txt [hunk 4]
```diff
@@ -60,6 +60,7 @@ Additional information on the Smartling Connector for WordPress can be found [he
 
 == Changelog ==
 = 1.16.0 =
+* Minimum required PHP version is now 7.2! The next plugin release will require PHP 7.4.
```
Comment (PavelLoparev) [inline readme.txt:63]: I believe we need to bump the major version of the plugin because this release is not compatible with previous one?

**Comment by PavelLoparev** [comment]
Seems like travis tests are failing

File: readme.txt (hunk 4 above)
Comment (vsolovei-smartling) [reply readme.txt:63]: It's fully compatible, customers are running it with 7.2 and 7.4 versions, it's just that older phpunit fails when using php 7 features

File: inc/Smartling/Helpers/WordpressContentTypeHelper.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/WordpressContentTypeHelper.php:41]: It's populated at runtime by wordpress, I've refactored \Smartling\Tests\Smartling\Base\SmartlingCoreTest::testExceptionOnTargetPlaceholderCreationFail to do without this change

File: inc/Smartling/Queue/Queue.php (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Queue/Queue.php:182]: Yes, a wordpress one, reverted the change, refactored affected tests

**Comment by vsolovei-smartling** [comment]
> Seems like travis tests are failing

Look like rate limit issue, last travis build was green

**Review by PavelLoparev — APPROVED** [review]

**Pull request merged**